Sources/TuistKit/Commands/DocCommand.swift

```diff
@@ -0,0 +1,77 @@
+import ArgumentParser
+import Foundation
+import TSCBasic
+import TuistSupport
+
+// MARK: - DocCommand
+
+struct DocCommand: ParsableCommand {
+    static var configuration: CommandConfiguration {
+        CommandConfiguration(commandName: "doc",
+                             abstract: "Generates html documentation for a given target.")
+    }
+
+    // MARK: - Options
+
+    @OptionGroup()
+    var options: DocCommand.Options
+
+    // MARK: - Run
+
+    func run() throws {
+        let absolutePath: AbsolutePath
+        if let path = options.path {
+            absolutePath = AbsolutePath(path, relativeTo: FileHandler.shared.currentPath)
+        } else {
+            absolutePath = FileHandler.shared.currentPath
+        }
+
+        try DocService().run(project: absolutePath,
+                             target: options.target,
+                             serve: options.mode.shouldServe,
+                             port: options.port)
+    }
+}
+
+// MARK: - Options
+
+extension DocCommand {
+    enum Mode: EnumerableFlag {
+        case localhost
+        case filesOnly
+
+        static func name(for value: DocCommand.Mode) -> NameSpecification {
+            switch value {
+            case .localhost:
+                return [.long]
+            case .filesOnly:
+                return [.long]
+            }
+        }
+
+        var shouldServe: Bool { self == .localhost }
+    }
+
+    struct Options: ParsableArguments {
+        @Option(
+            name: .shortAndLong,
+            help: "The path to the Project.swift container folder.",
+            completion: .directory
+        )
+        var path: String?
+
+        @Flag(
+            help: "Provide the documentation as md files in a temporal folder or serve it as a website."
+        )
+        var mode: DocCommand.Mode = .localhost
```

Comment:
I'd remove this flag. Let's default to opening the documentation in the browser. If it turns out that developers would find this feature valuable, we can add it. The wider the API surface is, the more user-paths we have to optimize for, and the more complex scenarios they will build upon Tuist.

Reply:
Makes sense. I still need a flag to avoid blocking the process when running the acceptance test. Is there any way to do it?